includes/Compatibility.php

```diff
@@ -0,0 +1,431 @@
+						);
+					$data = [
+						'title' => esc_html__( 'Web Stories plugin could not be initialized.', 'web-stories' ),
+					];
+					$this->error->add( 'failed_build', $message, $data );
+
+					return false;
+				}
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check to see if all required PHP extensions are installed.
+	 *
+	 * @return bool
+	 */
+	public function check_extensions() {
+		$_web_stories_missing_extensions = [];
+		foreach ( $this->get_extensions() as $_web_stories_required_extension => $_web_stories_required_constructs ) {
+			if ( ! extension_loaded( $_web_stories_required_extension ) ) {
+				$_web_stories_missing_extensions[] = "<code>$_web_stories_required_extension</code>";
+			}
+		}
+
+		if ( count( $_web_stories_missing_extensions ) > 0 ) {
+			$this->error->add(
+				'missing_extension',
+				sprintf(
+				/* translators: %s is list of missing extensions */
+					_n(
+						'The following PHP extension is missing: %s. Please contact your host to finish installation.',
+						'The following PHP extensions are missing: %s. Please contact your host to finish installation.',
+						count( $_web_stories_missing_extensions ),
+						'web-stories'
+					),
+					implode( ', ', $_web_stories_missing_extensions )
+				)
+			);
+
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check to see if classes exist.
+	 *
+	 * @return bool
+	 */
+	public function check_classes() {
+		$_web_stories_missing_classes = [];
+		foreach ( $this->get_extensions() as $_web_stories_required_extension => $_web_stories_required_constructs ) {
+			foreach ( $_web_stories_required_constructs as $_web_stories_construct_type => $_web_stories_constructs ) {
+				if ( 'classes' !== $_web_stories_construct_type ) {
+					continue;
+				}
+
+				foreach ( $_web_stories_constructs as $_web_stories_construct ) {
+					if ( ! class_exists( $_web_stories_construct ) ) {
+						$_web_stories_missing_classes[] = "<code>$_web_stories_construct</code>";
+					}
+				}
+			}
+		}
+
+		if ( count( $_web_stories_missing_classes ) > 0 ) {
+			$this->error->add(
+				'missing_class',
+				sprintf(
+				/* translators: %s is list of missing extensions */
+					_n(
+						'The following PHP class is missing: %s. Please contact your host to finish installation.',
+						'The following PHP classes are missing: %s. Please contact your host to finish installation.',
+						count( $_web_stories_missing_classes ),
+						'web-stories'
+					),
+					implode( ', ', $_web_stories_missing_classes )
+				)
+			);
+
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check to see if all require functions exist.
+	 *
+	 * @return bool
+	 */
+	public function check_functions() {
+		$_web_stories_missing_functions = [];
+		foreach ( $this->get_extensions() as $_web_stories_required_extension => $_web_stories_required_constructs ) {
+			foreach ( $_web_stories_required_constructs as $_web_stories_construct_type => $_web_stories_constructs ) {
```

Comment:
Since these are not global variables, can we just make them a bit shortr.

e.g. `$missing_functions` instead of `$_web_stories_missing_functions`, `$required_constructs` instead of `$_web_stories_required_constructs`, and so forth.

Same for `check_classes()` above

Reply:
This was as a copy and paste from the AMP plugin. Fixed in https://github.com/google/web-stories-wp/pull/4930/commits/70f7297c387c5c2a9696c8c203cc0705048a40ee